iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/TieLineAdderImpl.java

```diff
@@ -182,8 +195,40 @@ public TieLineImpl add() {
         if (ucteXnodeCode == null) {
             throw new ValidationException(this, "ucteXnodeCode is not set");
         }
-        checkHalf(half1, 1);
-        checkHalf(half2, 2);
```

Comment:
I would check if both half1 and half2 adders are set before creating half1 and half2